RosettaImmutables/src/main/java/com/hubspot/rosetta/immutables/ContextualHelper.java

```diff
@@ -48,7 +49,10 @@ private static JavaType findWireSafeTypeParameter(JavaType type) {
       return null;
     } else if (type.hasRawClass(WireSafeEnum.class)) {
       return type;
-    } else if (type.hasRawClass(Optional.class) || type.hasRawClass(com.google.common.base.Optional.class)) {
+    } else if (type.hasRawClass(Optional.class)
+        || type.hasRawClass(com.google.common.base.Optional.class)
+        || type.hasRawClass(List.class)
```

Comment:
We sould probably support all collection (and maybe Iterable?) types here. We'll also want to support all specific sub-implementations. When using `@HubSpotImmutableStyle` guava's immutable collection are required.

we should probably also support `Map` keys and values.

Reply:
I don't disagree that more collection and related types should be supported here, but I want to be cognizant of not scope creeping more to what this PR was intended to support and unblock our team on with Lists. I'd be happy to add those changes in follow up PRs if that suits?

Reply:
I think the root issue with all of this is that the serializer is using the BeanProperty. The deserializer uses the contextual type which is always going to be WireSafeEnum<X>, and doesn't have this problem. I think the BeanProperty is broken by design, because if you have a field like `Map<WireSafeEnum<X>, WireSafeEnum<Y>>`, which type should the serializer use? Luckily it seems like the serializer doesn't actually _need_ the type and was just using it for generating an error message

Reply:
I think `Map` could be scope creep, but supporting collection types beyond only and exactly `List` is something I'd want to see if we were moving forward with this PR.

Reply:
That's fair. In this case it seems like https://github.com/HubSpot/Rosetta/pull/84 might cover support for the remainder of the types.